appdevguide/deviceSupport.md

```diff
@@ -0,0 +1,524 @@
+
+This same pattern should be followed for all record types except for the {doc}`lsi <epics-base:lsiRecord>`,
+{doc}`lso <epics-base:lsoRecord>` and {doc}`printf <epics-base:printfRecord>` record types, 
+which have published their device support entry
+table structures since they were first added to Base but didn't previously embed
+the `dset common` member. Device support for these record types therefore can't
+use the dset name since the new definitions are different from the originals and
+will cause a compile error, so this pattern should be used instead:
+
+```C
+#ifndef HAS_lsidset
+struct {
+    dset common;
+    long (*read_string)(lsiRecord *prec);
+}
+#else
+lsidset
+#endif
+devLsiEtherIP = {
+    {5, NULL, lsi_init, lsi_init_record, get_ioint_info},
+    lsi_read
+};
+```
+
+The field `DTYP` contains the index of the menu choice as defined by the device ASCII definitions. `iocInit` uses this 
+field and the device support structures defined in {doc}`epics-base:devSup_h` to initialize the field `DSET`. 
+Thus record support can locate  its associated device support via the `DSET` field.
+
+Device support modules can be divided into two basic classes: synchronous and asynchronous. Synchronous device 
+support is used for hardware that can be accessed without delays for I/O. Many register based devices are synchronous 
+devices. Other devices, for example all GPIB devices, can only be accessed via I/O requests that may take large amounts 
+of time to complete. Such devices must have associated asynchronous device support. Asynchronous device support 
+makes it more difficult to create databases that have linked records.
+
+If a device can be accessed with a delay of less than a few microseconds then synchronous device support is appropriate. 
+If a device causes delays of greater than 100 microseconds then asynchronous device support is appropriate. If the delay is 
+between these values your guess about what to do is as good as mine. Perhaps you should ask the hardware designer why 
+such a device was created.
+
+If a device takes a long time to accept requests there is another option than asynchronous device support. A driver can be 
+created that periodically polls all its attached input devices. The device support just returns the latest polled value. For 
+outputs, device support just notifies the driver that a new value must be written. the driver, during one of its polling 
+phases, writes the new value. The EPICS Allen Bradley device/driver support is a good example.
+
+##  Example Synchronous Device Support Module
+
+```C
+/* Create the dset for devAiSoft */
+long init_record();
+long read_ai();
+struct {
+    long   number;
+    DEVSUPFUN   report;
+    DEVSUPFUN   init;
+    DEVSUPFUN   init_record;
+    DEVSUPFUN   get_ioint_info;
+    DEVSUPFUN   read_ai;
+    DEVSUPFUN   special_linconv;
+}devAiSoft={
+    6,
+    NULL,
+    NULL,
+    init_record,
+    NULL,
+    read_ai,
+    NULL
+};
```

Comment:
I know this would be more work, but it bothers me a little that we're publishing an example here that hasn't been updated following the changes described in the beginning section of the document. A naïve user might expect that the examples given would follow our latest recommendations and copy/paste them directly into new device support.

If you don't have time or want to wait for that to be done before publishing this, please file an issue reminding us to make those changes later.

Reply:
I do not mind updating the example. In this process I have not been sure if I should first just port the chapters from the old version as is and then hope that occasionally somebody else updates, or should I try to update myself while converting. The first case does not seem to happen a lot so I guess I will try to update the example.

Reply:
Should have been done when the change was introduced in the code 100 years ago, obviously. Thanks for wiping up.

Reply:
Just for fun, I looked up the example in the documentation...I could not find exactly this version, but in this shape it is essentially from Base 3.13, latest edit in this version by Janet Anderson, the second latest by Marty in 1998.
I guess it is time to update. The question would be if I again take a verbatim copy or try to minimize. I tend towards a full copy, because it illustrates some, ehh..., "new" features that often come up in discussions.

Reply:
I had a great idea to use the example device support that one gets with makeBaseApp.pl (devXxxSoft.c). That is where most new (and old)comers start, right? However, what do I get? The old interface, DSET copied in the source file.
TIme for an update, maybe. Or maybe I got something wrong.